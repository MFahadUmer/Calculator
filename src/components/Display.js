import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ calculate }) => (
  <div className="display">
    <p className="calculateText">{ calculate }</p>
  </div>
);

Display.propTypes = {
  calculate: PropTypes.string,
};

Display.defaultProps = {
  calculate: '0',
};

export default Display;
