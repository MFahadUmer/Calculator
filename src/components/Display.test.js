import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Display from './Display';

describe('Display Component Tests', () => {
  it('should display text ', () => {
    render(<Display calculate="15" />);
    expect(screen.getByText('15')).toBeTruthy();
  });

  it('should display text ', () => {
    render(<Display calculate="20" />);
    expect(screen.getByText('20')).toBeTruthy();
  });

  it('matches the snapshot of display component', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
