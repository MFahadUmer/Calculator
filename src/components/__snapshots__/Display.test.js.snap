// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component Tests matches the snapshot of display component 1`] = `
<div
  className="display"
>
  <p
    className="calculateText"
  >
    0
  </p>
</div>
`;
